controller/api/public/grpc_server.go

```diff
@@ -128,6 +132,32 @@ func (s *grpcServer) ListPods(ctx context.Context, req *pb.Empty) (*pb.ListPodsR
 	return resp, nil
 }
 
+func (s *grpcServer) SelfCheck(ctx context.Context, in *common.SelfCheckRequest) (*common.SelfCheckResponse, error) {
+	telemetryClientCheck := &common.CheckResult{
+		SubsystemName:    TelemetryClientSubsystemName,
+		CheckDescription: TelemetryClientCheckDescription,
+		Status:           string(healthcheck.CheckError),
+	}
+
+	_, err := s.telemetryClient.ListPods(ctx, &telemPb.ListPodsRequest{})
+	if err != nil {
+		telemetryClientCheck.Status = string(healthcheck.CheckError)
+		telemetryClientCheck.FriendlyMessageToUser = fmt.Sprintf("Error talking to telemetry service from control plane: %s", err.Error())
+		return nil, err
+	} else {
+		telemetryClientCheck.Status = string(healthcheck.CheckOk)
+	}
```

Comment:
Hmm, this conditional seems problematic. If `err != nil`, then you could just `return nil, err`. There's no need to modify any of the fields on `telemetryClientCheck`, since, as far as I can tell, you're discarding it. Maybe this return statement doesn't belong?